# 📐 T-Test in Statistics
### Hypothesis Testing using T-Score | Sheryians AI School

---

> **T-Test** is used to determine whether the **sample mean** is significantly different from the **population mean**, when the population standard deviation is **unknown** and we only have the **sample standard deviation**.

---

## 🤔 Z-Test vs T-Test — What's the Difference?

| | Z-Test | T-Test |
|---|---|---|
| Population Std Dev | ✅ Known | ❌ Unknown |
| Sample Size | Large (n ≥ 30) | Small (n < 30) |
| Distribution | Normal | T-Distribution |

---

## 📌 Problem Statement

A researcher claims that the **average height of adults** is **170 cm**.
We collect a sample of **10 individuals** but we **don't know** the population standard deviation.

| Detail | Value |
|---|---|
| Population Mean (μ) | 170 cm |
| Sample Size (n) | 10 |
| Significance Level (α) | 0.05 |

## 🔢 T-Test Formula

$$t = \frac{\bar{x} - \mu}{s / \sqrt{n}}$$

- $\bar{x}$ = Sample Mean  
- $\mu$ = Population Mean  
- $s$ = Sample Standard Deviation *(ddof=1)*  
- $n$ = Sample Size

In [1]:
# Importing required libraries
import numpy as np
from scipy import stats

## 📋 Step 1 — Define Sample & Population Parameters

We define our sample of heights. Notice — this time we use **sample standard deviation** (ddof=1), not population std dev.

> **ddof=1** means Degrees of Freedom = n-1, which gives us an unbiased estimate of the population std dev from our sample.

In [2]:
sample = [172, 174, 168, 169, 171, 173, 175, 170, 169, 172]

sample_mean = np.mean(sample)
sample_std = np.std(sample, ddof=1)   # ddof=1 for sample standard deviation
n = len(sample)
mean_population = 170

print(f"Sample        : {sample}")
print(f"Sample Mean   : {sample_mean}")
print(f"Sample Std Dev: {sample_std:.4f}")
print(f"Sample Size   : {n}")
print(f"Pop Mean      : {mean_population}")

Sample        : [172, 174, 168, 169, 171, 173, 175, 170, 169, 172]
Sample Mean   : 171.3
Sample Std Dev: 2.3118
Sample Size   : 10
Pop Mean      : 170


## 🔍 Step 2 — Calculate the T-Score

The T-Score tells us how many standard errors our sample mean is away from the population mean.

$$t = \frac{\bar{x} - \mu}{s / \sqrt{n}}$$

In [3]:
t_stats = (sample_mean - mean_population) / (sample_std / np.sqrt(n))

print(f"T-Score : {t_stats:.4f}")

T-Score : 1.7782


## 📊 Step 3 — Calculate the P-Value

The **P-Value** is calculated using the **T-Distribution** with **degrees of freedom = n - 1**.

- We use a **two-tailed test** because we're checking if the mean is *different* (not just higher or lower)
- **Degrees of Freedom (df) = n - 1 = 9**

> The T-Distribution has heavier tails than the Normal Distribution, which accounts for the extra uncertainty of not knowing the population std dev.

In [4]:
p_value = 2 * (1 - stats.t.cdf(abs(t_stats), df=n-1))
alpha = 0.05

print(f"P-Value            : {p_value:.6f}")
print(f"Alpha              : {alpha}")
print(f"Degrees of Freedom : {n-1}")

P-Value            : 0.109078
Alpha              : 0.05
Degrees of Freedom : 9


## ✅ Step 4 — Decision: Accept or Reject Null Hypothesis?

### Our Hypotheses:
- **H₀ (Null Hypothesis):** Average height = 170 cm *(no significant difference)*
- **H₁ (Alternative Hypothesis):** Average height ≠ 170 cm *(significant difference exists)*

### Decision Rule:

| Condition | Decision |
|---|---|
| p_value < α (0.05) | ❌ Reject H₀ |
| p_value ≥ α (0.05) | ✅ Accept H₀ |

In [5]:
print(f"T-Score  : {t_stats:.4f}")
print(f"P-Value  : {p_value:.6f}")
print(f"Alpha    : {alpha}")
print("-" * 40)

if p_value < alpha:
    print("❌ Reject the Null Hypothesis")
    print("→ Sample mean is significantly different from population mean.")
else:
    print("✅ Accept the Null Hypothesis")
    print("→ No significant difference found. The claim holds true.")

T-Score  : 1.7782
P-Value  : 0.109078
Alpha    : 0.05
----------------------------------------
✅ Accept the Null Hypothesis
→ No significant difference found. The claim holds true.


## 📝 Conclusion

| Metric | Value |
|---|---|
| Sample Mean | 171.3 cm |
| Population Mean | 170 cm |
| Sample Std Dev | ~2.36 |
| T-Score | 1.7782 |
| P-Value | ~0.1092 |
| Degrees of Freedom | 9 |
| Alpha (α) | 0.05 |
| Decision | ✅ Accept H₀ |

---

**Interpretation:**
Since **p-value (~0.1092) > alpha (0.05)**, we **fail to reject** the null hypothesis.
There is **not enough statistical evidence** to say the average height differs from 170 cm.

---

## 🔁 Z-Test vs T-Test — Final Comparison

| | Z-Test (Previous) | T-Test (This Notebook) |
|---|---|---|
| Std Dev Used | Population (σ = 3) | Sample (s ≈ 2.36) |
| Z / T Score | 1.3703 | 1.7782 |
| P-Value | 0.1706 | ~0.1092 |
| Decision | ✅ Accept H₀ | ✅ Accept H₀ |

> 💡 **Key Takeaway:** T-Test gives a **higher test statistic and lower p-value** than Z-Test on the same data — because it accounts for the extra uncertainty of estimating std dev from the sample.

